core/src/createTextMaskInputElement.js

```diff
@@ -9,14 +9,62 @@ const strNone = 'none'
 
 export default function createTextMaskInputElement({
   inputElement,
-  mask: providedMask,
+  mask,
   guide,
-  pipe,
+  pipe: providedPipe,
   placeholderChar = defaultPlaceholderChar,
   onAccept,
   onReject,
   keepCharPositions = false
 }) {
+  // Anything that we will need to keep between `update` calls, we will store in this `state` object.
+  const state = {previousConformedValue: emptyString}
+
+  const {
+    mask: providedMask,
+    placeholder,
+    pipe
+  } = processMaskAndPlaceholder({
+    mask,
+    pipe: providedPipe,
+    placeholderChar
+  })
+
+  return {
+    state,
+    // `update` is called by framework components whenever they want to update the `value` of the input element.
+    // The caller can send a `rawValue` to be conformed and set on the input element. However, the default use-case
+    // is for this to be read from the `inputElement` directly.
+    update: function(rawValue = inputElement.value) {
+      // `selectionStart` indicates to us where the caret position is after the user has typed into the input
+      const {selectionStart: currentCaretPosition} = inputElement
+
+      const {value = '', adjustedCaretPosition} = getConformedInputState({
+        rawValue,
+        previousConformedValue: state.previousConformedValue,
+        guide,
+        pipe,
+        mask: providedMask,
+        placeholderChar,
+        onAccept,
+        onReject,
+        placeholder,
+        keepCharPositions,
+        currentCaretPosition,
+      })
+
+      inputElement.value = value // set the input value
+      safeSetSelection(inputElement, adjustedCaretPosition) // adjust caret position
+      state.previousConformedValue = value // store value for access for next time
+    }
+  }
+}
+
+export function processMaskAndPlaceholder({
+  mask: providedMask,
```

Comment:
Was just matching the previous code. Also looked weird when I did `mask = mask.mask`. Happy to change it though